[bug] NDK path with spaces doesn't work in CMakeToolchain

### Describe the bug

macOS 14.7, Conan v2.9.2

If NDK path contains spaces, then it's inserted in the toolchain file as is and cmake doesn't like it of course. Example:

    conan install -c tools.android:ndk_path=/Volumes/Android\ NDK\ r27c/AndroidNDK12479018.app/Contents/NDK ...

then at cmake configure time there'll be error like:

```
CMake Error at conan-android64-25/conan_toolchain.cmake:39 (include):
  include called with invalid argument:
  r27c/AndroidNDK12479018.app/Contents/NDK/build/cmake/android.toolchain.cmake
Call Stack (most recent call first):
  /Users/kambala/dev/vcmi/build-android64/CMakeFiles/3.31.0/CMakeSystem.cmake:6 (include)
  CMakeLists.txt:6 (project)
```

Embedding quotes inside the path also doesn't help:

```
conan install -c tools.android:ndk_path='"/Volumes/Android NDK r27c/AndroidNDK12479018.app/Contents/NDK"' ...
...
CMake Error at conan-android64-25/conan_toolchain.cmake:39 (include):
  include requested file is a directory:

    /Volumes/Android NDK r27c/AndroidNDK12479018.app/Contents/NDK
Call Stack (most recent call first):
  /Users/kambala/dev/vcmi/build-android64/CMakeFiles/3.31.0/CMakeSystem.cmake:6 (include)
  CMakeLists.txt:6 (project)
```

Adding the config to a profile instead results in exactly the same issues.

### How to reproduce it

pass `-c tools.android:ndk_path='/some/path with spaces'` to `conan install` and check the generated toolchain file or try to configure a CMake project with it

--- a/conan/tools/cmake/toolchain/blocks.py
+++ b/conan/tools/cmake/toolchain/blocks.py
@@ -375,7 +375,7 @@ class AndroidSystemBlock(Block):
         {% if android_use_legacy_toolchain_file %}
         set(ANDROID_USE_LEGACY_TOOLCHAIN_FILE {{ android_use_legacy_toolchain_file }})
         {% endif %}
-        include({{ android_ndk_path }}/build/cmake/android.toolchain.cmake)
+        include("{{ android_ndk_path }}/build/cmake/android.toolchain.cmake")
         """)
 
     def context(self):

--- a/test/integration/toolchains/cmake/test_cmaketoolchain.py
+++ b/test/integration/toolchains/cmake/test_cmaketoolchain.py
@@ -831,7 +831,7 @@ def test_android_c_library():
     conan_toolchain = client.load(os.path.join(client.current_folder, "conan_toolchain.cmake"))
     assert "set(ANDROID_PLATFORM android-23)" in conan_toolchain
     assert "set(ANDROID_ABI x86_64)" in conan_toolchain
-    assert "include("/foo/build/cmake/android.toolchain.cmake")" in conan_toolchain
+    assert 'include("/foo/build/cmake/android.toolchain.cmake")' in conan_toolchain
     client.run("create . --name=foo --version=1.0 " + settings)
 
 

--- a/test/integration/toolchains/cmake/test_cmaketoolchain.py
+++ b/test/integration/toolchains/cmake/test_cmaketoolchain.py
@@ -831,7 +831,7 @@ def test_android_c_library():
     conan_toolchain = client.load(os.path.join(client.current_folder, "conan_toolchain.cmake"))
     assert "set(ANDROID_PLATFORM android-23)" in conan_toolchain
     assert "set(ANDROID_ABI x86_64)" in conan_toolchain
-    assert "include(/foo/build/cmake/android.toolchain.cmake)" in conan_toolchain
+    assert "include("/foo/build/cmake/android.toolchain.cmake")" in conan_toolchain
     client.run("create . --name=foo --version=1.0 " + settings)
 
 

